content/terraform/v1.13.x/docs/language/meta-arguments/depends_on.mdx

```diff
@@ -0,0 +1,178 @@
+
+`check` blocks validate infrastructure outside of the typical resource lifecycle. You can nest a `data` block inside a `check` block to fetch information that you can reference in an `assert` block. By adding a `depends_on` argument to the `data` block nested in a `check` block requires, Terraform retrieves the necessary data before running the validation.
+
+For example, if you define a `check` that verifies that a website API returns `200`, that check fails the first time Terraform runs your configuration because your website's infrastructure does not exist yet. You can set the `depends_on` argument to a resource, such as the load balancer, to ensure Terraform only runs the `check` once the website is up. When running an operation, Terraform evaluates the `check`, warns `known after apply` until that crucial piece of your website is ready, and continues the operation.
+
+However, this strategy only works when the `data` block does not directly reference the resource specified in the `depends_on` argument. Otherwise, anytime that resource changes, the `check` block warns `known after apply` until Terraform updates that resource, making your check potentially noisy and ineffective. 
+
+In the following example, Terraform waits to run the check until it creates the `aws_db_instance.main` database. Terraform prints `known after apply`, instead of printing false warnings, until it finishes creating the database:
+
+```hcl
+check "database_connection" {
+ data "postgresql_database" "app_db" {
+   name       = "application"
+   depends_on = [aws_db_instance.main]
+ }
+
+ assert {
+   condition     = data.postgresql_database.app_db.allow_connections
+   error_message = "Database is not accepting connections"
+ }
+}
+```
+
+### Specify a dependency when querying a data source
+
+In the following example, Terraform gets an AMI for use in the `aws_instance` resource after creating the `aws_subnet.example_subnet` resource:
+
+```hcl
+resource "aws_vpc" "example_vpc" {
+  cidr_block = "10.0.0.0/16"
+}
+
+resource "aws_subnet" "example_subnet" {
+  vpc_id     = aws_vpc.example_vpc.id
+  cidr_block = "10.0.1.0/24"
+}
+
+data "aws_ami" "example_ami" {
+  most_recent = true
+  owners      = ["amazon"]
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm-*"]
+  }
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+  depends_on = [aws_subnet.example_subnet]
+}
+
+resource "aws_instance" "example_instance" {
+  ami           = data.aws_ami.example_ami.id
+  instance_type = "t2.micro"
+  subnet_id     = aws_subnet.example_subnet.id
+}
+```
+
+Note that this is a simplified example and does not represent a real-world use case. The `depends_on` argument changes the dependency graph so that Terraform creates the resources in a linear order. Without the argument, Terraform would still successfully create the resources.
```

Comment:
Should we use this example if it's not a real use case? Let's find a real one instead